src/java/fr/paris/lutece/plugins/appointment/web/AppointmentApp.java

```diff
@@ -333,6 +334,12 @@ public synchronized XPage getViewAppointmentCalendar( HttpServletRequest request
         {
             // If we want to change the date of an appointment
             AppointmentDTO appointmentDTO = AppointmentService.buildAppointmentDTOFromRefAppointment( refAppointment );
+            // Check if an appointment Object was found and built with the given reference
+            if( appointmentDTO == null )
+            {
+            	// When the appointment Object doesn't exist, we display an error message to the user
```

Comment:
1/ why not just use getXpage method ? 

 model.put( MARK_NO_APPOINTMENT_WITH_THIS_REFERENCE, Boolean.TRUE );
 return getXPage( TEMPLATE_ERROR_APPOINTMENT_REFERENCE, getLocale( request ), model );
 
 
2/ what's the use of the boolean MARK_NO_APPOINTMENT_WITH_THIS_REFERENCE ? it's not used in the template ?


3/ to display a simple message, SiteMessageService is maybe more appropriate ?

 SiteMessageService.setMessage( request, MESSAGE_xxx, SiteMessage.TYPE_xxxx, theurl.getUrl( ) );
 return null;

Reply:
Indeed, this process can be simplified. It will be tested with SiteMessageService